devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,298 @@
+```
+
+**SRS_INTERLOCKED_43_014 [** `interlocked_exchange` shall set a 32-bit variable to the specified value as an atomic operation.**]**
+
+## interlocked_exchange_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_exchange_16, volatile int16_t*, target, int16_t, value);
+```
+
+**SRS_INTERLOCKED_43_015 [** `interlocked_exchange_16` shall set a 16-bit variable to the specified value as an atomic operation.**]**
+
+
+## interlocked_exchange_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_exchange_64, volatile int64_t*, target, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_016 [** `interlocked_exchange_64` shall set a 64-bit variable to the specified value as an atomic operation.**]**
+
+## interlocked_exchange_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_exchange_8, volatile int8_t*, target, int8_t, value);
+```
+
+**SRS_INTERLOCKED_43_017 [** `interlocked_exchange_8` shall set an 8-bit variable to the specified value as an atomic operation.**]**
+
+## interlocked_exchange_add
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_exchange_add, volatile int32_t*, addend, int32_t, value);
+```
+
+**SRS_INTERLOCKED_43_018 [** `interlocked_exchange_add` shall perform an atomic addition of two 32-bit values.**]**
+
+## interlocked_exchange_add_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_exchange_add_64, volatile int64_t*, addend, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_019 [** `interlocked_exchange_add_64` shall perform an atomic addition of two 64-bit values.**]**
+
+## interlocked_exchange_pointer
+
+```c
+MOCKABLE_FUNCTION(, void*, interlocked_exchange_pointer, volatile void*, target, void*, value);
+```
+
+**SRS_INTERLOCKED_43_020 [** `interlocked_exchange_pointer` shall atomically exchange a pair of addresses.**]**
+
+## interlocked_increment
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_increment, volatile int32_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_021 [** `interlocked_increment` shall increment (increase by one) the value of the specified 32-bit variable as an atomic operation.**]**
+
+## interlocked_increment_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_increment_16, volatile int16_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_022 [** `interlocked_increment_16` shall increment (increase by one) the value of the specified 16-bit variable as an atomic operation.**]**
+
+## interlocked_increment_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_increment_64, volatile int64_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_023 [** `interlocked_increment_64` shall increment (increase by one) the value of the specified 64-bit variable as an atomic operation.**]**
+
+## interlocked_or
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_or, volatile int32_t*, destination, int32_t, value);
+```
+
+**SRS_INTERLOCKED_43_024 [** `interlocked_or` shall perform an atomic OR operation on the specified 32-bit integer values. The function prevents more than one thread from using the same variable simultaneously.**]**
```

Comment:
>The function prevents more than one thread from using the same variable simultaneously [](start = 123, length = 86)

Isn't this the case for every of the interlocked_ APIs? I would not mention it here, but someplace where the module is explained.